src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -282,6 +282,20 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
     return this.cleanParams(params);
   }
 
+  protected buildRequestsHttpOptions(): any {
```

Comment:
let's make it optional?

Reply:
This makes sense since clientId and clientSecret are optional

Reply:
Sorry, my comment was a bit misleading. What I meant is that probably we should add an option to the configuration as you proposed in the PR description indicating if we want to enable the base auth, pass clientid/cliensecret as body parameters or none of the above?

Reply:
@alain-charles just checking if you saw my comment :)